model/src/main/java/uk/gov/hmcts/fees2/register/data/model/RangedFee.java

```diff
@@ -44,17 +44,22 @@ public boolean isInRange(BigDecimal amount) {
 
     @Override
     public boolean isADuplicateOf(Fee anotherFee) {
+        if(!super.isADuplicateOf(anotherFee)) {
```

Comment:
I guess it might fail if we compare with mixed type fees. I mean, Fee to insert is "FixedType" but DB has RangedFee with same reference data including keyword. Can you double check please?

Reply:
The super function compares classes already

Reply:
Yes, but that wouldn't protect the above scenario. We shouldn't allow different fee types with same reference data (including keyword).

Reply:
I see your point now, I dont know why i was thinking it was the other way around